party_time/party_time.lua

```diff
@@ -0,0 +1,341 @@
+    check = function(str)
+        local bool = arg_lookups.bool[str]
+        if bool ~= nil then
+            return bool
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.bool, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+command.arg.register_type('lookup_add_remove', {
+    check = function(str)
+        local op = arg_lookups.addremove[str]
+        if op ~= nil then
+            return op
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.addremove, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+-- Addon Command Handlers
+-- Seven main commands additoinal sub_commands within
+local pt = command.new('pt')
+
+local invite = function(...)
+    for _, name in pairs({...}) do
+        command.input('/pcmd add ' .. name)
+        name = name:gsub('^%l', string.upper)
+        if unhandled_requests[name] then
+            unhandled_requests[name] = nil
+        end
+    end
+end
+
+pt:register('i', invite, '{name}*')
+pt:register('invite', invite, '{name}*')
+
+
+local request = function(name)
+    command.input('/prcmd add ' .. name)
+end
+
+pt:register('r', request, '{name}')
+pt:register('request', request, '{name}')
+
+
+local blacklist = function(sub_cmd, ...)
+    local names = sets({...})
+    options.blacklist[sub_cmd](options.blacklist, names)
+    settings.save(options)
+end
+
+pt:register('b', blacklist, '<sub_cmd:lookup_add_remove> {name}*')
+pt:register('blacklist', blacklist, '<sub_cmd:lookup_add_remove> {name}*')
+
+
+local whitelist = function(sub_cmd, ...)
+    local names = sets({...})
+    options.whitelist[sub_cmd](options.whitelist, names)
+    settings.save(options)
+end
+
+pt:register('w', whitelist, '<sub_cmd:lookup_add_remove> {name}*')
+pt:register('whitelist', whitelist, '<sub_cmd:lookup_add_remove> {name}*')
+
+
+local ui_enable = function(bool)
+    options.ui.enabled = bool
+    settings.save(options)
+end
+
+pt:register('ui_enable', ui_enable, '<enabled:lookup_boolean>')
+
+
+local auto_accept_enable = function(bool)
+    options.auto.accept = bool
+    settings.save(options)
+end
+
+pt:register('auto_accept', auto_accept_enable, '<enabled:lookup_boolean>')
+
+
+local auto_decline_enable = function(bool)
+    options.auto.decline = bool
+    settings.save(options)
+end
+
+pt:register('auto_decline', auto_decline_enable, '<enabled:lookup_boolean>')
+
+-- Packet Event Handlers
+-- Recieve Invite & Recieve Request
+packets.incoming[0x0DC]:register(function(p)
+    if options.auto.accept and options.whitelist[p.player_name] then
+        coroutine.schedule(function()
+            local clock = os.clock()
+            repeat
+                if (os.clock() - clock) > 90 then
+                    return
+                end
+                coroutine.sleep_frame()
+            until(#treasure == 0)
+            command.input('/join')
+        end)
+    elseif options.auto.decline and options.blacklist[p.player_name] then
```

Comment:
I would test if this still works, because I don't think it does. It should now be `options.blacklist:contains(p.player_name)`. Similarly for the whitelist below.

Set items cannot be checked with `[]`, which is done by design, since it would return invalid values in certain cases. As an example:
```lua
local empty_set = sets({})
if empty_set['union'] then
    print('This should not show, but it does')
end
```